documentation/api/query/v4/operators.markdown

```diff
@@ -171,10 +177,73 @@ To get the average uptime for your nodes,
 ## Subquery Operators
 
 Subqueries allow you to correlate data from multiple sources or multiple
-rows. (For instance, a query such as "fetch the IP addresses of all nodes with
-`Class[Apache]`" would have to use both facts and resources to return a list of facts.)
+rows. For instance, a query such as "fetch the IP addresses of all nodes with
+`Class[Apache]`" would have to use both facts and resources to return a list of facts.
+
+There are two forms of subqueries, implicit and explicit, and both forms work the
+same under the hood. The implicit form however, only requires you to specify the
+related entity, while the explicit form requires you to be specify exactly how
+data should be joined during the subquery.
+
+### `subquery` (Implicit Subqueries)
+
+*Note:* Implicit subqueries are a new experimental feature, be warned the functionality
+may change as we provide improvements.
+
+Implicit queries work like most operators, and simply require you to specify the
+related entity and the query to use:
+
+    ["subquery", "<ENTITY>", <SUBQUERY STATEMENT>]
+
+The entity is a mapping to the particular endpoint or data type you desire, however not
+all entities are supported from all endpoints, as not every relationship makes sense.
+Consult the endpoint document for details on what implicit relationships are supported:
+
+* [catalogs][`catalogs`]
+* [edges][`edges`]
+* [environments][`environments`]
+* [events][`events`]
+* [facts][`facts`]
+* [fact-contents][`fact_contents`]
+* [fact-paths][`fact_paths`]
+* [nodes][`nodes`]
+* [reports][`reports`]
+* [resources][`resources`]
+
+Internal to PuppetDB, we keep a mapping of how different entities relate to each
+other, and so no other data beyond the entity is needed in this case. This is
+different to explicitly subqueries, for those you must specify yourself how
+two entities are related, although functionally they can produce the same results.
 
-Subqueries are unlike the other operators listed above. They always appear together in the following form:
+#### Implicit Subquery Examples
+
+A query string like the following on the [nodes][`nodes`] endpoint will return the list
+of all nodes with the `Package[Tomcat]` resource in its catalog, and a certname starting
+with `web1`:
+
+    ["and",
+      ["~", "certname", "^web1"],
+      ["subquery", "resources",
+        ["=", "type", "Package"],
+        ["=", "title", "Tomcat"]]]
+
+If you wanted to display the entire `networking` fact, if the hosts interfaces uses a certain mac address
+you can do the following on the [facts][`facts`] endpoint:
+
+    ["and",
+      ["=", "name", "networking"]
+      ["subquery", "fact_contents",
+        ["~>", "path", ["networking", ".*", "macaddresses", ".*"]],
+        ["=", "value", "aa:bb:cc:dd:ee:00"]]]
```

Comment:
this needs a comma after the closing bracket on networking, but it also doesn't for me when pasted:


```
[~/work/puppetdb] (ticket/master/pdb-1924-implicit-subqueries) $ curl -X GET http://localhost:8080/pdb/query/v4/facts -d 'query=["and", ["=", "name", "networking"], ["subquery", "fact_contents", ["~>", "path", ["networking", ".*", "macaddresses", ".*"]], ["=", "value", "aa:bb:cc:dd:ee:00"]]]'
Query operator ~> is not allowed on field path
```

not sure what the story is. I haven't read far enough to know yet but it also seems like the subquery clauses of this one and the one above should contain an "and".